File: 1147-longest-chunked-palindrome-decomposition/1147-longest-chunked-palindrome-decomposition.py
class Solution(object):
    def longestDecomposition(self, s):
        """
        :type text: str
        :rtype: int
        """
        res, n, l, r = 0, len(s), "", ""
        for i in xrange(n):
            l += s[i]
            r = s[n - i - 1] + r
            if l == r:
                res += 1
                l, r = "", ""
        return res

File: 1147-longest-chunked-palindrome-decomposition/1147-longest-chunked-palindrome-decomposition.cpp
class Solution {
public:
    int longestDecomposition(string s) {
        int res = 0; int n = s.length();
        string l = ""; string r = "";
        for(int i = 0; i < n; ++i) {
            l += s[i]; r = s[n - i - 1] + r;
            if(l == r) {
                res++;
                l = "";
                r = "";
            }
        }
        return res;
    }
};